user_guide_src/source/installation/upgrade_4xx.rst

```diff
@@ -1,114 +1,102 @@
+We'll try to point out the most important considerations here.
+
+Not all of the CI3 libraries have been ported or rewritten for CI4!
+See the threads in the `CodeIgniter 4 Roadmap <https://forum.codeigniter.com/forum-33.html>`_
+subforum for an uptodate list!
+
+**Downloads**
+- CI4 is still available as a ready-to-run zip or tarball, which
+includes the user guide (though in the `docs` subfolder
+- It can also be installed using Composer
+
+**Namespaces**
+- CI4 is built for PHP7.1+, and everything in the framework is namespaced, even the helpers
+
+**Application Structure**
+- The framework still has `application` and `system` folders, with the same 
+interpretation as before
+- The framework now provides for a `public` folder, intended as the document
+root for your app
+- There is also a `writable` folder, to hold cache data, logs, and session data
+- The `application` folder looks very similar to that for CI3, with some
+name changes (eg Filters instead of hooks) and capitalization, and some subfolders
+moved to the `writable` folder
+- There is no longer a nested `application/core` folder, as we have
+a different mechanism for extending framework components (see below)
+
+**Class loading**
+- There is no longer a CodeIgniter superobject, with framework component
+references magically as properties of your controller
+- Classes are instantiated where needed, and components are managed
+by `Services`
+- The class loader automatically handles PSR4 style class locating,
+within the `App` (application) and `CodeIgniter` (i.e. system) top level
+namespaces; with composer autoloading support, and even using educated
+guessing to find your models and libraries if they are in the right
+folder even though not namespaced
+- You can configure the class loading to support whatever application structure
+you are most comfortable with, including the "HMVC" style
+
+**Controllers**
+- Controllers extend \CodeIgniter\Controller instead of CI_Controller
+- They don't use a constructor any more (to invoke CI "magic"); they
+instead call `initController`
+- CI provides `Request` and `Response` objects for you to work with -
+more powerful than the CI3-way
+- If you want a base controller (MY_Controller in CI3), make it
+where you like, e.g. BaseController extends Controller, and then
+have your controllers extend it
+
+**Models**
+- Models extend \CodeIgniter\Model instead of CI_Model
+- The CI4 model has much more functionality, including automatic
+database connection, basic CRUD, in-model validation, and
+automatic pagination
+- CI4 also has the `Entity` class you can build on, for
+richer data mapping to your database tables
+- Instead of CI3's `$this->load->model(x);`, you would now use
+`$this->x = new X();`, providing the fully-qualified path
+of your component
+
+**Views**
+- Your views look much like before, but they are invoked differently ...
+instead of CI3's `$this->load->view(x);` you can use `view(x);`
```

Comment:
You'll need to do `echo view()` not just `view()`

Reply:
Fixing :)